tests/reporter/kubernetes.go

```diff
@@ -114,20 +114,24 @@ func (r *KubernetesReporter) Report(report types.Report) {
 
 func (r *KubernetesReporter) ReportSpec(specReport types.SpecReport) {
 	fmt.Fprintf(GinkgoWriter, "On failure, artifacts will be collected in %s/%d_*\n", r.artifactsDir, r.failureCount+1)
-	if !r.alwaysReport && r.failureCount > r.maxFails {
+	if !r.programmaticFocus && r.failureCount > r.maxFails {
 		return
 	}
 	if specReport.Failed() {
 		r.failureCount++
-	} else if !r.alwaysReport {
+	} else if !r.programmaticFocus {
 		return
 	}
 
 	// If we got not directory, print to stderr
 	if r.artifactsDir == "" {
 		return
 	}
-	By("Collecting Logs for failed test")
+	reason := "due to failure"
+	if r.programmaticFocus {
```

Comment:
It actually is a good idea to emit the reason in this case :+1:

However, I'd want to change the reason to the what rather here

```suggestion
	if r.programmaticFocus {
	         reason = "due to debug logging enabled"
```

WDYT?

Reply:
While this is tied to the specific use of focus containers I'd rather be explicit here.